builders/integrations/wallets/rainbow-kit.md

```diff
@@ -0,0 +1,209 @@
+
+Next, add a file called `.env.local` to your root project directory to store your WalletConnect `projectId`. Ensure your project contains a `.gitignore` file in this same directory and `.env*.local` is added to the list of files to ignore. Add your `projectId` to the `.env.local` file as follows:
+
+`NEXT_PUBLIC_PROJECT_ID='INSERT_PROJECT_ID'`
+
+In the next section, you will use this stored `projectId` when setting up the `wagmi` config.
+
+## Connect DApp to MetaMask Mobile {: #connect-dapp-to-metamask-mobile }
+
+In this section, you will complete the needed steps to use RainbowKit to connect your dApp to MetaMask's mobile wallet:
+
+1. Import RainbowKit, Wagmi, and TanStack Query
+2. Setup configuration for Wagmi
+3. Wrap your application with providers
+4. Add the connect button
+
+### Import RainbowKit, Wagmi, and TanStack Query
+
+Ensure you are still in your project's root directory, then create a new file called `wagmi.ts`. This file will contain the imports and configuration needed to connect your dApp to mobile wallets and interact with blockchains. 
+
+Start by adding the imports for RainbowKit, Wagmi, and TanStack Query. Make a note of the list of chains imported from `wagmi/chains`.  Wagmi provides an extensive list of [supported chains](https://wagmi.sh/core/api/chains#:~:text=core/chains%27-,Available,-Chains){target=\_blank} you can reference to locate the expected names for Moonbeam ecosystem chains. 
+
+```js title="wagmi.ts"
+--8<-- 'code/builders/integrations/wallets/rainbowkit/rainbow-imports.js'
+```
+
+### Configure Wagmi to Support Moonbase Alpha 
+
+Next, configure the supported chains and setup a `wagmi` config .  Add `moonbaseAlpha` to the `wagmi/chains` import and the `chains` array in the config to add it to the list of your dApp's supported networks.
+
+```js title="wagmi.ts"
+--8<-- 'code/builders/integrations/wallets/rainbowkit/wagmi-config.js'
+```
+
+### Wrap Your Application with Providers
+
+With the configuration in place, the next step is to wrap your application with the `RainbowKitProvider`, `WagmiProvider`, and `QueryClientProvider` to make them available throughout your dApp. Inside the `app` directory, create a file named `providers.tsx` and add the following code to define `Providers`:
+
+```js title="providers.tsx"
+--8<-- 'code/builders/integrations/wallets/rainbowkit/providers.ts'
+```
+
+Now open the `layout.tsx` file and modify the code to import `Providers` and wrap the application:
+
+```js title="layout.tsx"
+--8<-- 'code/builders/integrations/wallets/rainbowkit/layout.tsx'
+```
+
+### Add the Connect Button
+
+RainbowKit offers a `ConnectButton` component, which renders the **Connect** and **Disconnect** buttons and UI elements for switching chains. This example imports the `ConnectButton` into the `page.tsx` file for simplicity, but you may want to add it to an element like a **Header** or **Navbar** so it appears at the top of each page. Update the code in `page.tsx` as follows:
```

Comment:
where is the page.tsx file? there haven't been instructions to add it anywhere yet